src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -625,13 +635,25 @@ public TResult TryGetLast(out bool found)
                 return sourceFound ? _selector(input) : default(TResult);
             }
 
+            private TResult[] LazyToArray()
```

Comment:
Why separate this out?  The other cases in ToArray aren't in their own methods, seems like this should just be inline with the others.